### PR TITLE
Method min

### PR DESCRIPTION
**Pull request by LuisARodr**
This adds the min(comparer) method functionality to the framework.
Also includes tests for the method.

Description of the method:
>This method will find and return the minimum value on the collection, a null value will be returned for empty arrays. If not comparer is specified then it will try to evaluate the array elements as if they are numbers. The comparator should be a function that receives 2 parameters, valueA and valueB and shall return a negative number for valueA < valueB scenario, zero when valueA === valueB and finally a positive number when valueA > valueB.

### CODE REVIEW

File: Methods/ArrayExtension.js [hunk 1]
```diff
@@ -355,6 +355,49 @@ Array.prototype.max = function(comparer) {
     return max;
 }
 
+/**
+ * Returns the minumum value on the collection or null if the array is empty,
+ * if comparer is not present it will evaluate the elements as if they where 
+ * numbers.
+ * @param {comparer} comparer - A function used to make the comparations, this
+ * param not being present means the comparations will be made as if every value
```
Comment (gustavostuff) [inline Methods/ArrayExtension.js]: comparison*

File: Methods/ArrayExtension.js [hunk 2]
```diff
@@ -355,6 +355,49 @@ Array.prototype.max = function(comparer) {
...
+ * on the array where numbers.
+ * @returns {*} - The minumum value on the collection acording to the comparer.
+ * @throws {TypeError} - If spec is present and is not a function.
```
Comment (gustavostuff) [inline Methods/ArrayExtension.js:366]: according

**Review by gustavostuff — CHANGES_REQUESTED** [review]

File: Methods/ArrayExtension.js (hunk 1 above)
Comment (LuisARodr) [reply Methods/ArrayExtension.js]: Fixed on a22fd8f

File: Methods/ArrayExtension.js (hunk 2 above)
Comment (LuisARodr) [reply Methods/ArrayExtension.js:366]: Fixed on a22fd8f

**Review by gustavostuff — APPROVED** [review]

**Comment by LuisARodr** [comment]
Closing to merge into [dev](https://github.com/LuisARodr/array-extension/tree/dev)

**Pull request closed without merging**